apis/sqs/v1beta1/queue_types.go

```diff
@@ -51,20 +49,19 @@ type RedrivePolicy struct {
 	// The Amazon Resource Name (ARN) of the dead-letter queue to which Amazon
 	// SQS moves messages after the value of maxReceiveCount is exceeded.
 	// +optional
-	DeadLetterQueueARN *string `json:"deadLetterQueueARN,omitempty"`
+	DeadLetterTargetARN *string `json:"deadLetterTargetArn,omitempty"`
 
-	// DeadLetterQueueARNRef reference a Queue to retrieve its ARN.
+	// DeadLetterTargetARNRef reference a Queue to retrieve its ARN.
 	// +optional
-	DeadLetterQueueARNRef *runtimev1alpha1.Reference `json:"deadLetterQueueArnRef,omitempty"`
+	DeadLetterTargetARNRef *runtimev1alpha1.Reference `json:"deadLetterTargetArnRef,omitempty"`
 
-	// DeadLetterQueueARNSelector selects reference to a Queue to retrieve its ARN
+	// DeadLetterTargetARNSelector selects reference to a Queue to retrieve its ARN
 	// +optional
-	DeadLetterQueueARNSelector *runtimev1alpha1.Selector `json:"deadLetterQueueArnSelector,omitempty"`
+	DeadLetterTargetARNSelector *runtimev1alpha1.Selector `json:"deadLetterTargetArnSelector,omitempty"`
 
 	// The number of times a message is delivered to the source queue before
 	// being moved to the dead-letter queue.
-	// +optional
-	MaxReceiveCount *int64 `json:"maxReceiveCount,omitempty"`
+	MaxReceiveCount int64 `json:"maxReceiveCount"`
```

Comment:
I am guessing that this is only required if the dead letter target is also?

Reply:
Yes, but `deadLetterTargetArn` is referenced so we can't mark it as required.

Reply:
Okay. I think we typically would remove the `// +optional` off of `deadLetterTargetArn` in this case then (i.e. when it is required but it is a reference). My concern here with `MaxReceiveCount` becoming required is that now this PR technically could break an existing implementation. It would be very unlikely, but if someone had tried to use `deadLetterQueueArn` and did not use `MaxReceiveCount` and it was ignored by AWS (which, judging by the error if the value was supplied and no `MaxReceiveCount` was supplied it looks like it _might_ ignore it, but maybe not) so they go a queue just with the `RedrivePolicy` missing, then their YAML that "worked" would now be in violation of the schema here.

It could be that AWS would reject this already anyway, in which case this is not an issue. Even if it doesn't the Queue would not be configured in the way the user expected. But I did want to surface the fact that a previously valid YAML could now be invalid, and we are not bumping versions.

Reply:
> if someone had tried to use deadLetterQueueArn and did not use MaxReceiveCount and it was ignored by AWS (which, judging by the error if the value was supplied and no MaxReceiveCount was supplied it looks like it might ignore it, but maybe not)

It's not ignored. I realized that during the manual test that if you send any kind of JSON blob for `redrivePolicy`, it has to include both `deadLetterTargetArn` and `maxReceiveCount`. Also keep in mind that there is no way YAML can work with `deadLetterQueueArn` field name. I don't know where that came from but AWS outright rejects it and expects `deadLetterTargetArn`.

Reply:
Gotcha, so the only break here is that a YAML that was being rejected by AWS is now rejected by the API server. That seems okay to me without version bump 👍 Thanks!

Reply:
Yeah. I was kind of on the fence about whether we should just expect a JSON blob as string like AWS but didn't go for that.